error.go

```diff
@@ -3,14 +3,26 @@ package gimlet
 import (
 	"fmt"
 	"net/http"
+
+	"github.com/mongodb/grip/message"
 )
 
-// APIError implements the Error() interface
+// ErrorResponse implements the Error() interface and provides a
+// common type for routes to use for returning payloads in error
+// conditions.
+//
+// ErrorResponse also implements grip's message.Composer interface
+// which can simplify some error reporting on the client side.
 type ErrorResponse struct {
-	StatusCode int    `json:"status" yaml:"status"`
-	Message    string `json:"error" yaml:"message"`
+	StatusCode   int    `bson:"status" json:"status" yaml:"status"`
+	Message      string `bson:"message" json:"message" yaml:"message"`
+	message.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
 }
 
 func (e ErrorResponse) Error() string {
 	return fmt.Sprintf("%d (%s): %s", e.StatusCode, http.StatusText(e.StatusCode), e.Message)
 }
+
+func (e ErrorResponse) String() string   { return e.Error() }
+func (e ErrorResponse) Raw() interface{} { return e }
+func (e ErrorResponse) Loggable() bool   { return e.StatusCode > 299 }
```

Comment:
Is this like an HTTP status code? If so, I would have expected `e.StatusCode > 399`.

Reply:
fixed